### ISSUE TITLE
error reading from Consul

### ISSUE THREAD

**Issue by wangnuannuan** [edgexfoundry/edgex-go#742]
1.  I clone the repo and then run `make build`,  three errors occurred in file `edgex-go/internal/export/distro/format.go` . When call `uuid.NewV4()` , for example at line 124, you should write it like `id, _ := uuid.NewV4()`.
2.  In file `edgex-go/cmd/support-logging/main.go` 

```
	flag.BoolVar(&useConsul, "consul", true, "Indicates the service should use consul.")
	flag.BoolVar(&useConsul, "c", true, "Indicates the service should use consul.")
```
I run the Consul agent use `./consul agent -dev`, and initialized database by using `init_mongo.js`( get from https://github.com/edgexfoundry/docker-edgex-mongo/raw/master/init_mongo.js)
and then an Error occurred `error reading from Consul`

```
func connectToConsul(conf *ConfigurationStruct) (*ConfigurationStruct, error) {
	//Obtain ConsulConfig
	cfg := consulclient.NewConsulConfig(conf.Registry, conf.Service, internal.SupportLoggingServiceKey)
	// Register the service in Consul
	err := consulclient.ConsulInit(cfg)

	if err != nil {
		return conf, fmt.Errorf("connection to Consul could not be made: %v", err.Error())
	}
	// Update configuration data from Consul
	updateCh := make(chan interface{})
	errCh := make(chan error)
	dec := consulclient.NewConsulDecoder(conf.Registry)
	dec.Target = &ConfigurationStruct{}
	dec.Prefix = internal.ConfigV2Stem + internal.SupportLoggingServiceKey
	dec.ErrCh = errCh
	dec.UpdateCh = updateCh

	defer dec.Close()
	defer close(updateCh)
	defer close(errCh)
	go dec.Run()

	select {
	case <-time.After(2 * time.Second):
		err = errors.New("timeout loading config from registry")
	case ex := <-errCh:
		err = errors.New(ex.Error())
	case raw := <-updateCh:
		actual, ok := raw.(*ConfigurationStruct)
		if !ok {
			return conf, errors.New("type check failed")
		}
		conf = actual
		fmt.Printf("Updated config form Consul: %#v\n", conf)
		//Check that information was successfully read from Consul
		if len(conf.Persistence) == 0 {
			return nil, errors.New("error reading from Consul")

		}
	}
	return conf, err
}
```
log
```
Updated config form Consul: &logging.ConfigurationStruct{Persistence:"", Databases:map[string]config.DatabaseInfo(nil), Logging:config.LoggingInfo{EnableRemote:false, File:"", Level:""}, Registry:config.RegistryInfo{Host:"", Port:0, Type:""}, Service:config.ServiceInfo{BootTimeout:0, CheckInterval:"", ClientMonitor:0, Host:"", Port:0, Protocol:"", StartupMsg:"", ReadMaxLimit:0, Timeout:0}}
ERROR: 2018/11/04 06:36:06 error reading from Consul
```

**Comment by tsconn23** [comment]
Did you run the config seed service to populate the configuration into Consul? That needs to be done before any service will bootstrap if starting services with the `--consul` flag. When the above error is thrown it most likely means there is no entry in Consul for the requested key. Assuming `make build` has been done, config-seed executable is found in `edgex-go/cmd/config-seed`

**Comment by jduranf** [comment]
@wangnuannuan, seems you are not using the proper uuid libraries. As it's explained in the [README.md](https://github.com/edgexfoundry/edgex-go#installation-and-execution), before running `make build`, please run `make prepare`

**Comment by tsconn23** [comment]
This panic
```
created by github.com/edgexfoundry/edgex-go/pkg/clients/general.(*generalRestClient).init
	/home/wang/Documents/go/src/github.com/edgexfoundry/edgex-go/pkg/clients/general/client.go:42 +0xe9
```
appears to be coming from the sys-mgmt-agent service. If I'm correct, there will be a fix in master shortly. It has been fixed in the Delhi branch as PR #727. Would advise you to just comment out `exec -a edgex-sys-mgmt-agent ./sys-mgmt-agent &` in the edgex-launch.sh for now. @akramtexas for visibility.

The ERROR: lines related to the schedule and schedule events are being addressed in Delhi by PR #740. Once that PR is verified, the fix will be brought into master. @xmlviking for visibility.

Neither of these will effect the primary workflow within EdgeX. If it bothers you, just comment out the respective lines to start the services from the shell script until the master branch PRs are ready.

Also, be sure you're adding `--consul` to the `exec` line for all of the services in the shell script.

**Comment by akramtexas** [comment]
@tsconn23 Yep, a fix will be in master shortly. It has been fixed in the Delhi branch as PR #727.

**Comment by akramtexas** [comment]
The fix (to which I had referred in the prior note) has been merged into the master branch per PR #746

**Comment by wangnuannuan** [comment]
I am trying to create MQTT device, and I have see the `device-sdk-go`.  What I need to do is create a driver like the `device-sdk-go/examples/simple/simpledriver.go`, right?

**Comment by jduranf** [comment]
Yes @wangnuannuan, but I recommend to wait a big PR which should be merged this week and will modify a lot `device-sdk-go`.
Also you can create the device using `device-sdk-c` or use `device-mqtt`, which is in java.

**Comment by tsconn23** [comment]
@wangnuannuan Feel free to take questions about device services to the [Rocket Chat](https://chat.edgexfoundry.org/home). I'm assuming the discussion above has answered your initial questions.

**Comment by wangnuannuan** [comment]
ok, thank you
